src/content_cao.cpp

```diff
@@ -1612,6 +1620,12 @@ void GenericCAO::processMessage(const std::string &data)
 		updateNodePos();
 	} else if (cmd == GENERIC_CMD_SET_TEXTURE_MOD) {
 		std::string mod = deSerializeString(is);
+
+		// immediatly reset a engine issued texture modifier if a mod sends a different one
+		if (m_reset_textures_timer > 0) {
+			m_reset_textures_timer = -1;
+			updateTextures(m_previous_texture_modifier);
+		}
```

Comment:
Probably better to do

```cpp
if (m_reset_textures_timer > 0) {
    m_reset_textures_timer = -1;
    m_current_texture_modifier = m_previous_texture_modifier
}
```

as the texture is immediately redone below, and the temp effect is lost due to the shifting